src/schema/contentPreference.ts

```diff
@@ -0,0 +1,285 @@
+import { followEntity, unfollowEntity } from '../common/contentPreference';
+import { GQLEmptyResponse, offsetPageGenerator } from './common';
+import graphorm from '../graphorm';
+import { Connection, ConnectionArguments } from 'graphql-relay';
+
+export type GQLContentPreference = Pick<
+  ContentPreference,
+  'referenceId' | 'userId' | 'type' | 'createdAt' | 'status'
+>;
+
+export const typeDefs = /* GraphQL */ `
+  ${toGQLEnum(ContentPreferenceType, 'ContentPreferenceType')}
+
+  ${toGQLEnum(ContentPreferenceStatus, 'ContentPreferenceStatus')}
+
+  enum FollowStatus {
+    follow
+    subscribed
+  }
+
+  type ContentPreference {
+    referenceId: ID!
+
+    user: User!
+
+    type: ContentPreferenceType!
+
+    createdAt: DateTime!
+
+    status: ContentPreferenceStatus!
+  }
+
+  type ContentPreferenceEdge {
+    node: ContentPreference!
+
+    """
+    Used in \`before\` and \`after\` args
+    """
+    cursor: String!
+  }
+
+  type ContentPreferenceConnection {
+    pageInfo: PageInfo!
+    edges: [ContentPreferenceEdge!]!
+  }
+
+  extend type Query {
+    """
+    Content preference status
+    """
+    contentPreferenceStatus(
+      """
+      Id of the entity
+      """
+      id: ID!
+      """
+      Entity type (user, source..)
+      """
+      entity: ContentPreferenceType!
+    ): ContentPreference @auth
+
+    """
+    Who follows user
+    """
+    userFollowers(
+      """
+      Id of user
+      """
+      userId: ID!
+      """
+      Entity to list (user, source..)
+      """
+      entity: ContentPreferenceType!
+      """
+      Paginate after opaque cursor
+      """
+      after: String
+      """
+      Paginate first
+      """
+      first: Int
+    ): ContentPreferenceConnection!
+
+    """
+    What user follows
+    """
+    userFollowing(
+      """
+      Id of user
+      """
+      userId: ID!
+      """
+      Entity to list (user, source..)
+      """
+      entity: ContentPreferenceType!
+      """
+      Paginate after opaque cursor
+      """
+      after: String
+      """
+      Paginate first
+      """
+      first: Int
+    ): ContentPreferenceConnection!
+  }
+
+  extend type Mutation {
+    """
+    Follow entity
+    """
+    follow(
+      """
+      Id of the entity
+      """
+      id: ID!
+      """
+      Entity to follow (user, source..)
+      """
+      entity: ContentPreferenceType!
+      """
+      Follow status
+      """
+      status: FollowStatus!
+    ): EmptyResponse @auth
+    """
+    Unfollow entity
+    """
+    unfollow(
+      """
+      Id of the entity
+      """
+      id: ID!
+      """
+      Entity unfollow (user, source..)
+      """
+      entity: ContentPreferenceType!
+    ): EmptyResponse @auth
+  }
+`;
+
+const contentPreferencePageGenerator =
+  offsetPageGenerator<GQLContentPreference>(10, 50);
+
+export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
+  unknown,
+  BaseContext
+>({
+  Query: {
+    contentPreferenceStatus: async (
```

Comment:
Just more wondering where do we actually use this singular call?

Reply:
In some cases like search page or post page author/scout we have singular follow buttons so might be easier to lazy load state like this instead of nested user resolve.

Reply:
Got it :)